include/secp256k1.h

```diff
@@ -40,6 +40,26 @@ extern "C" {
 #  define SECP256K1_ARG_NONNULL(_x)
 # endif
 
+# if defined _MSC_VER || defined __CYGWIN__
+#  define SECP256K1_HELPER_DLL_IMPORT __declspec(dllimport)
+#  define SECP256K1_HELPER_DLL_EXPORT __declspec(dllexport)
+# else
+#  if __GNUC__ >= 4
+#   define SECP256K1_HELPER_DLL_IMPORT __attribute__ ((visibility ("default")))
+#   define SECP256K1_HELPER_DLL_EXPORT __attribute__ ((visibility ("default")))
+#  else
+#   define SECP256K1_HELPER_DLL_IMPORT
+#   define SECP256K1_HELPER_DLL_EXPORT
+#  endif
+# endif
+
+# if defined SECP256K1_STATIC
+#  define SECP256K1_API
+# elif defined SECP256K1_DLL
```

Comment:
Sounds good to me.

I think this should be ```__attribute__ ((visibility ("hidden")))``` in the ```SECP256K1_STATIC``` case though, since it wouldn't need to be externally visible